### PR TITLE
Align time range to interval

### LINKED ISSUE #3781: Align  The requested period with Interval if Interval templating is used
Currently Grafana  uses many intervals based on the current time, for example at 13:34:56   asking for last 24 hours will look at the interval  from 13:34:56 from the last day. 

When uses has Interval templating enabled one would expect information to be aligned on the interval boundary,  For example if  I am looking at last 24 hours  with Interval set to 1 hour  I would rather see data from  14:00:00 last day to 14:00:00   not have potentially misleading information for the first interval in the day. 

This is especially valuable for accumulated value - for example if I am looking at avg network throughput it does not matter that much if hour is complete.  If I'm looking to see how much traffic in GB I'm  producing per hour viewing full hours would be preferred.

### PR DESCRIPTION
**Pull request by mtanda**
Try to fix https://github.com/grafana/grafana/issues/3781 (labeled "P3: important")

### CODE REVIEW

File: public/app/features/panel/metrics_panel_ctrl.ts [hunk 1]
```diff
@@ -166,6 +166,23 @@ class MetricsPanelCtrl extends PanelCtrl {
       return this.$q.when([]);
     }
 
+    var targets = JSON.stringify(this.panel.targets);
+    var intervalVar = null;
+    _.each(this.templateSrv.variables, function(variable) {
```
Comment (torkelo) [inline public/app/features/panel/metrics_panel_ctrl.ts]: This feels very hacks,  can you describe what your doing and why, also is there a better way?

Comment (mtanda) [inline public/app/features/panel/metrics_panel_ctrl.ts]: Align time range cause extra data points fetch which isn't plotted on screen.
So, avoid such fetch for the panels which doesn't use "interval" variable.

Comment (mtanda) [inline public/app/features/panel/metrics_panel_ctrl.ts]: The JSON format of targets is differed among data source plugin, so I use JSON.stringify() to check query expression easily.

**Comment by torkelo** [comment]
@mtanda will you look at this PR https://github.com/grafana/grafana/pull/4257 ? 

I think this can be redone so that you only check interval and not JSON stringify all queries and look for use of variables.

https://github.com/mtanda/grafana/blob/issue_3781/public/app/features/panel/metrics_panel_ctrl.ts#L116

**Comment by mtanda** [comment]
@torkelo OK, I'll reconsider for this issue after PR is merged.

**Pull request closed without merging**